jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -625,9 +634,74 @@ def _update_stretch_histogram(self, msg={}):
             # in case only the sample has changed but its length has not,
             # we'll force the traitlet to trigger a change
             hist_mark.send_state('sample')
-
         self._stretch_histogram_needs_update = False
 
+    @observe('stretch_vmin_value', 'stretch_vmax_value', 'layer_selected',
+             'stretch_function_value', 'stretch_hist_nbins',
+             'image_contrast_value', 'image_bias_value')
+    def _update_stretch_curve(self, msg=None):
+        # Import here to prevent circular import (and not at the top of the method so the import
+        # check is avoided, whenever possible).
+        from jdaviz.configs.imviz.plugins.viewers import ImvizImageView
+        from jdaviz.configs.cubeviz.plugins.viewers import CubevizImageView
+
+        if (
+            not isinstance(self.viewer.selected_obj, (ImvizImageView, CubevizImageView)) or
+            not hasattr(self, 'stretch_histogram')
+        ):
+            # don't update histogram if selected viewer is not an image viewer,
+            # or the stretch histogram hasn't been initialized:
+            return
+
+        mark_label_prefix = "stretch_curve: "
+        for layer in self.layer.selected_obj:
+            # clear old mark, if it exists:
+            mark_label = f'{mark_label_prefix}{layer.label}'
+            mark_exists = mark_label in self.stretch_histogram.marks
+
+            # create the new/updated mark following the colormapping
+            # procedure in glue's CompositeArray:
+            layer_state = layer.state.as_dict()

```

Comment:
does this provide any benefits over referring to `layer.state.contrast` etc below?  My only thought would be that you ensure all state properties are accessed simultaneously 🤷‍♂️

Reply:
No important differences. Switching to `layer.state.`.